SilentPatchSA/FLACDecoderSA.cpp

```diff
@@ -1,159 +1,43 @@
-	if ( processedChannels > 1 )
-	{
-		std::copy_n( buffer[1], pClientData->m_curBlockSize, it );
-	}
-
-	return FLAC__STREAM_DECODER_WRITE_STATUS_CONTINUE;
-}
-
-void CAEFLACDecoder::meta_cb(const FLAC__StreamDecoder* decoder, const FLAC__StreamMetadata *metadata, void *client_data)
-{
-	UNREFERENCED_PARAMETER(decoder);
-	if ( metadata->type == FLAC__METADATA_TYPE_STREAMINFO )
-	{
-		// Cache the header
-		CAEFLACDecoder*	pClientData = static_cast<CAEFLACDecoder*>(client_data);
-		pClientData->m_streamMeta = FLAC__metadata_object_clone(metadata);
-		assert( pClientData->m_streamMeta != nullptr );
-	}
-}
-
-FLAC__StreamDecoderSeekStatus CAEFLACDecoder::seek_cb(const FLAC__StreamDecoder *decoder, FLAC__uint64 absolute_byte_offset, void *client_data)
-{
-	UNREFERENCED_PARAMETER(decoder);
-	CAEFLACDecoder*	pClientData = static_cast<CAEFLACDecoder*>(client_data);
-	LARGE_INTEGER	li;
-	li.QuadPart = absolute_byte_offset;
-
-	return SetFilePointerEx(pClientData->GetStream()->GetFile(), li, nullptr, FILE_BEGIN) != 0 ? FLAC__STREAM_DECODER_SEEK_STATUS_OK : FLAC__STREAM_DECODER_SEEK_STATUS_ERROR;
-}
-
-FLAC__StreamDecoderTellStatus CAEFLACDecoder::tell_cb(const FLAC__StreamDecoder *decoder, FLAC__uint64 *absolute_byte_offset, void *client_data)
-{
-	UNREFERENCED_PARAMETER(decoder);
-	CAEFLACDecoder*	pClientData = static_cast<CAEFLACDecoder*>(client_data);
-	LARGE_INTEGER	li;
-	li.QuadPart = 0;
-
-	BOOL result = SetFilePointerEx(pClientData->GetStream()->GetFile(), li, &li, FILE_CURRENT);
-	*absolute_byte_offset = li.QuadPart;
-
-	return result != 0 ? FLAC__STREAM_DECODER_TELL_STATUS_OK : FLAC__STREAM_DECODER_TELL_STATUS_ERROR;
-}
-
-FLAC__StreamDecoderLengthStatus	CAEFLACDecoder::length_cb(const FLAC__StreamDecoder *decoder, FLAC__uint64 *stream_length, void *client_data)
-{
-	UNREFERENCED_PARAMETER(decoder);
-	CAEFLACDecoder*	pClientData = static_cast<CAEFLACDecoder*>(client_data);
-	LARGE_INTEGER	li;
-
-	BOOL bResult = GetFileSizeEx(pClientData->GetStream()->GetFile(), &li);
-	*stream_length = li.QuadPart;
-
-	return bResult ? FLAC__STREAM_DECODER_LENGTH_STATUS_OK: FLAC__STREAM_DECODER_LENGTH_STATUS_ERROR;
-}
-
-FLAC__bool CAEFLACDecoder::eof_cb(const FLAC__StreamDecoder* decoder, void* client_data)
-{
-	UNREFERENCED_PARAMETER(decoder);
-	CAEFLACDecoder*	pClientData = static_cast<CAEFLACDecoder*>(client_data);
-	return pClientData->m_eof;
-}
-
-void CAEFLACDecoder::error_cb(const FLAC__StreamDecoder* decoder, FLAC__StreamDecoderErrorStatus status, void* client_data)
-{
-	// Not implemented
-	UNREFERENCED_PARAMETER(decoder);
-	UNREFERENCED_PARAMETER(status);
-	UNREFERENCED_PARAMETER(client_data);
-}
-
 
 bool CAEFLACDecoder::Initialise()
 {
-	m_FLACdecoder = FLAC__stream_decoder_new();
-	assert( m_FLACdecoder != nullptr );
-	if ( m_FLACdecoder == nullptr )
-		return false;
+	DWORD size;
+	BOOL result = ReadFile(GetStream()->GetFile(), m_buffer, sizeof(m_buffer), &size, nullptr);
```

Comment:
This reads 4 bytes into the pointer `m_buffer` and then uses that as a pointer for `drflac_open_memory` - is this intentional?